map/api_mark_container.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "user_mark_container.hpp"
+#include "api_mark_point.hpp"
+
+class ApiUserMarkContainer : public UserMarkContainer
+{
+public:
+  ApiUserMarkContainer(double layerDepth, Framework & framework);
+
+  virtual Type GetType() const { return API_MARK; }
+
+  virtual string GetActiveTypeName() const;
+
+protected:
+  virtual string GetTypeName() const;
```

Comment:
Please, remove virtual and add override instead. Moreover, for groups of overridden methods (in public and protected section) add following comment:
```c++
// UserMarkContainer overrides:
Type GetType() const override { ... }
string GetActiveTypeName() const override { ... }
...
```